aggregation/options.go

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package aggregation
+
+import "github.com/m3db/m3metrics/policy"
+
+var (
+	defaultOptions = Options{UseDefaultAggregation: true, HasExpensiveAggregations: false}
+)
+
+// Options is the options for aggregations.
+type Options struct {
+	// UseDefaultAggregation means only default aggregation types are enabled.
+	UseDefaultAggregation bool
+	// HasExpensiveAggregations means expensive (multiplication／division)
+	// aggregation types are enabled.
+	HasExpensiveAggregations bool
+}
+
+// NewOptions creates a new aggregation options.
+func NewOptions() Options {
+	return defaultOptions
+}
+
+// ResetSetData resets the aggregation options.
+func (o *Options) ResetSetData(aggTypes policy.AggregationTypes) {
```

Comment:
Yes, which is intended, if the default/custom aggregation types for this timer does not want stdev or sumsq, HasExpensiveOperations will be reset to false and skip the sumsq calculation.